### ISSUE TITLE
[BUG] Referencing `@property` fields in extractor `prompt_template` throws error

### ISSUE THREAD

**Issue by barapa** [Mirascope/mirascope#200]
Example:

```python
from typing import Type

from mirascope.anthropic.extractors import AnthropicExtractor
from mirascope.anthropic.types import AnthropicCallParams


class TestPropertyExtractor(AnthropicExtractor[int]):
    extract_schema: Type[int] = int

    call_params = AnthropicCallParams(
        model="claude-3-haiku-20240307",
        temperature=0.0,
        max_tokens=100,
    )

    @property
    def input_text(self) -> str:
        return "Give me a number between 1 and 1000"

    prompt_template = """
    SYSTEM:
    The user is going to give you some instructions. You better follow them!

    USER:
    {input_text}
    """


if __name__ == "__main__":
    extractor = TestPropertyExtractor()
    result = extractor.extract()
    print(result)
```

Logs:
```bash
Traceback (most recent call last):
  File "/Users/br/development/projects/legaide/test-properties.py", line 31, in <module>
    result = extractor.extract()
             ^^^^^^^^^^^^^^^^^^^
  File "/Users/br/development/projects/legaide/.venv/lib/python3.12/site-packages/mirascope/anthropic/extractors.py", line 88, in extract
    return self._extract(AnthropicCall, AnthropicTool, retries, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/br/development/projects/legaide/.venv/lib/python3.12/site-packages/mirascope/base/extractors.py", line 136, in _extract
    response = TempCall(**self.model_dump(exclude={"extract_schema"})).call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/br/development/projects/legaide/.venv/lib/python3.12/site-packages/mirascope/anthropic/calls.py", line 60, in call
    messages, kwargs, tool_types = self._setup_anthropic_kwargs(kwargs)
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/br/development/projects/legaide/.venv/lib/python3.12/site-packages/mirascope/anthropic/calls.py", line 214, in _setup_anthropic_kwargs
    messages = self.messages()
               ^^^^^^^^^^^^^^^
  File "/Users/br/development/projects/legaide/.venv/lib/python3.12/site-packages/mirascope/anthropic/calls.py", line 46, in messages
    return self._parse_messages(
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/br/development/projects/legaide/.venv/lib/python3.12/site-packages/mirascope/base/prompts.py", line 143, in _parse_messages
    content = self._format_template(match.group(2))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/br/development/projects/legaide/.venv/lib/python3.12/site-packages/mirascope/base/prompts.py", line 98, in _format_template
    attr = getattr(self, var)
           ^^^^^^^^^^^^^^^^^^
  File "/Users/br/development/projects/legaide/.venv/lib/python3.12/site-packages/pydantic/main.py", line 792, in __getattr__
    raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}')
AttributeError: 'TempCall' object has no attribute 'input_text'
```

**Comment by willbakst** [comment]
Ah, good catch! Fix coming shortly just writing tests

**Comment by willbakst** [comment]
@barapa this is now released in fix version `v0.11.1` :)
